src/govuk/components/character-count/character-count.test.js

```diff
@@ -2,12 +2,21 @@
  * @jest-environment puppeteer
  */
 
+const jestPuppeteerConfig = require('../../../../jest-puppeteer.config.js')
 const { getExamples } = require('../../../../lib/jest-helpers.js')
 const { goToComponent, renderAndInitialise } = require('../../../../lib/puppeteer-helpers')
 
+// Detect when browser has been launched headless
+const { launch: { headless = true } } = jestPuppeteerConfig
```

Comment:
Absolutely, I'll change it 👍

Reply:
@romaricpascal Hope you don't mind a single (with default) destructured property?

Otherwise it defaults to `undefined` as it's not in the launch config.

```mjs
// Detect when browser has been launched headless
const { headless = true } = jestPuppeteerConfig.launch
```

Reply:
How do we feel about the [Nullish coalescing operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_operator)?

```js
// Detect when browser has been launched headless
const headless = jestPuppeteerConfig.launch.headless ?? true;
```

Reply:
Yeah, it was having the two levels of destructuring that I think is too much (not sure if we can lint for that, but that's a separate issue).

Reply:
That's not my understanding of how `??` works…

```
node
Welcome to Node.js v16.17.0.
Type ".help" for more information.
> const config = { truthy: true, falsey: false, nullish: null }
undefined
> config.truthy ?? true
true
> config.falsey ?? true
false
> config.nullish ?? true
true
> config.undefined ?? true
true
```

Reply:
Sorry I used the assignment form `??=` above

Depends if we want to guard against config values?

```mjs
launch: { headless: 0 }
launch: { headless: '' }
launch: { headless: NaN }
```

Reply:
Personally, I'm not too concerned about `launch.headless` being something unexpected as it's in our control, and find the coalesced version easier to understand than the destructured version. But recognise this is probably down to a lack of familiarity with how de-structuring with defaults works and basically personal preference, so this isn't blocking.

Reply:
Thanks @36degrees, appreciate the feedback

I suppose it's deciding when something's ready to be added to the code style arsenal?

```mjs
const hello = example.hello // Yes
const hello = example['hello'] // Yes
const { hello } = example // Maybe?
```

With all the clever Sass we do, it's nice to have the same artistic licence in JavaScript too 🎩 

```mjs
// Start with three
const more = {
  one: 1,
  two: 2,
  three: 3
}

// Take one out
const { one, ...less } = more

// Then there were two
console.log(less)

// { two: 2, three: 3 }
```

(That would be a bit too clever)